src/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -24,13 +18,32 @@ export interface DropdownMenuProps {
     className?: string;
 }
 
-export const DropdownMenu = ({ dropdownMenuItems, username, className }: DropdownMenuProps) => {
+export const DropdownMenu = ({ dropdownMenuItems, className }: DropdownMenuProps) => {
     const [isOpen, setIsOpen] = useState(false);
+    const wixApi = useContext(WixAPIContext);
+    const [username, setUsername] = useState<string | null>(null);
+
+    const onLoginClick = async () => {
+        if (username) {
+            await wixApi.logout();
+            setUsername(null);
+        } else {
+            await wixApi.initiateLogin();
+        }
+    };
+
+    useEffect(() => {
+        wixApi.fetchUserAuthData().then((response) => {
+            if (response) {
+                setUsername(response?.user?.member?.profile?.nickname || "");
```

Comment:
and then here it can be `|| username`

Reply:
I just realized it could be `|| username` anyway, since `setUsername` can accept null too as a value.